src/Microsoft.Extensions.DependencyInjection.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -50,45 +73,93 @@ public static class ServiceCollectionExtensions
                 throw new ArgumentNullException(nameof(implementationFactory));
             }
 
-            return Add(collection, service, implementationFactory, ServiceLifetime.Transient);
+            return Add(services, serviceType, implementationFactory, ServiceLifetime.Transient);
         }
 
-        public static IServiceCollection AddScoped(
-            this IServiceCollection collection,
-            Type service,
-            Type implementationType)
+        /// <summary>
+        /// Adds a transient service of the type specified in <typeparamref name="TService"/> with an
+        /// implementation type specified in <typeparamref name="TImplementation"/> to the
+        /// specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <typeparam name="TService">The type of the service to add.</typeparam>
+        /// <typeparam name="TImplementation">The type of the implementation to use.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        /// <seealso cref="ServiceLifetime.Transient"/>
+        public static IServiceCollection AddTransient<TService, TImplementation>(this IServiceCollection services)
+            where TService : class
+            where TImplementation : class, TService
         {
-            if (collection == null)
+            if (services == null)
             {
-                throw new ArgumentNullException(nameof(collection));
+                throw new ArgumentNullException(nameof(services));
             }
 
-            if (service == null)
+            return services.AddTransient(typeof(TService), typeof(TImplementation));
+        }
+
+        /// <summary>
+        /// Adds a transient service of the type specified in <paramref name="serviceType"/> to the
+        /// specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+        /// <param name="serviceType">The type of the service to register.</param>
```

Comment:
Maybe this would be clearer as "The type of the service to register and the implementation to use."